### PR TITLE
[ github-actions ] : Add workflow to get preview from fork repository and deploy to cf pages

### LINKED ISSUE #330: Get previews for PRs from forks
#### Describe the bug
Currently, it seems that preview sites are not generated for PRs from forks.  This is annoying.

#### How to Reproduce
Example: https://github.com/json-schema-org/website/pull/314

#### Expected behavior
It should generate a preview site, like [this](https://github.com/json-schema-org/website/pull/277#issuecomment-1922165611).

### PR DESCRIPTION
**Pull request by aialok**
**What kind of change does this PR introduce?**
- This PR introduces two workflows: one for preview deployment and another for production.
- We are transitioning from native Cloudflare app deployment to our own manual workflow for both preview and production deployments.
- Since Cloudflare does not support previews from forked repositories, we are implementing manual deployment via GitHub Actions.

**Issue Number:**

- Closes #330 

**Screenshots/videos:**

**If relevant, have you updated the documentation?**

**Summary**
- For complete details check out this comment : https://github.com/json-schema-org/website/issues/330#issuecomment-2115743948

**Does this PR introduce a breaking change?**

### CODE REVIEW

File: .github/workflows/preview-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+               node-version: 20
+        
+            - name: Install dependencies
+              run: yarn install --frozen-lockfile
+
+            - name: Build Site
+              run: yarn run build
+
+            - name: Deploy to Cloudflare Pages
+              id: cloudflare-pages-deploy
```
Comment (AdrianGonz97) [inline .github/workflows/preview-deployment.yml:77]:
```suggestion
```

adding an `id` isn't necessary if you're not referencing the output in another step

**Review by AdrianGonz97 — APPROVED** [review]
Thanks for the consideration of using my action. Everything looks good to me!

All that's left to do is to:
1. [generate an API key](https://github.com/AdrianGonz97/refined-cf-pages-action#generate-an-api-token) and [add the Cloudflare credentials](https://github.com/AdrianGonz97/refined-cf-pages-action#add-cloudflare-credentials-to-github-secrets) as env secrets
1. [remove the existing github integration](https://github.com/AdrianGonz97/refined-cf-pages-action#disabling-the-cloudflare-pages-github-integration) so that they don't conflict

---

I also want to reiterate what's said on the [README](https://github.com/json-schema-org/website/blob/72414490bf7fac5b937b8a415a25c244caccb31e/.env.example#L1-L2) regarding secrets and  PR previews from forks:

If _private secrets_ are being used during *build time* (which doesn't seem to be the case here) or at *runtime* for preview deployments  (I can't tell from a glance, but is viewable from `Workers & Pages > YourSiteName > Settings > Environment Variables`), then [PR approvals](https://github.com/AdrianGonz97/refined-cf-pages-action#enabling-pr-previews-from-forks) should be added so that each deployment requires approval. 

Taking a look at `.env.example`: https://github.com/json-schema-org/website/blob/72414490bf7fac5b937b8a415a25c244caccb31e/.env.example#L1-L2

it doesn't seem like any private secrets are being used, making preview deployments without pre-approval OK.

**Comment by aialok** [comment]
Thank you @AdrianGonz97 for your time and consideration.

I will be making the requested changes : )

**Review by benjagm — APPROVED** [review]
Let's try this! Great job Alok!

**Pull request merged**